### PR TITLE
add a generic Tile transformation; like numpy.tile

### PR DESCRIPTION
**Pull request by FynnBe**
Not sure about the preferred way to add a dimension agnostic transformation.

This `class Tile(torch.nn.Module)` implementation does not inherite from a kornia class (as there is no public common base class for 2D/3D), but is compatible with them.. maybe there should be an abstract base class for these cases...

### CODE REVIEW

File: environment_cpu.yaml [hunk 1]
```diff
@@ -5,10 +5,11 @@ channels:
 name:
     torch-em-cpu
 dependencies:
+    - affogato
     - cpuonly
     - elf
-    - affogato
     - pip
+    - pytest
```
Comment (constantinpape) [inline environment_cpu.yaml:12]: I would rather not mix testing frameworks, can you use unittest?

**Comment by constantinpape** [comment]
> This `class Tile(torch.nn.Module)` implementation does not inherite from a kornia class (as there is no public common base class for 2D/3D), but is compatible with them.. maybe there should be an abstract base class for these cases...

Could you give an example of how you use the `Tile` transform, and how you pass it to the corresponding dataset?
I think that would help to decide whether to add a base class for this.

**Comment by FynnBe** [comment]
> Could you give an example of how you use the Tile transform, and how you pass it to the corresponding dataset?
I think that would help to decide whether to add a base class for this.

Initially I wanted something like:
```python
    raw_transform = torch_em.transform.get_raw_transform(
        augmentation1=torch_em.transform.EMDefectAugmentation(p_drop_slice=p_drop_slice),
        augmentation2=torch_em.transform.Tile(reps=(2, 1, 1, 1, 1), match_shape_exactly=False),
    )
    label_transform = torch_em.transform.Tile(reps=(2, 1, 1, 1, 1), match_shape_exactly=False)  # todo: expand/braodcast_to instead of tile

    loader = torch_em.default_segmentation_loader(
        ...,
        raw_transform=raw_transform,
        label_transform=label_transform,
        transform=...,
        ...
)
```
-> creating two versions of the same input due to independent transformations following `Tile`

But I think I'll do something more explicit/manual instead... Still thought the `Tile` could come in handy some time.

**Comment by FynnBe** [comment]
Instead of a general transformation base class a [runtime checkable protocol](https://docs.python.org/3/library/typing.html#typing.runtime_checkable) would probably the proper way to go to be compatible with kornia, without inheriting from the protected `kornia.augmentation.base._AugmentationBase`

**Comment by constantinpape** [comment]
I don't really understand the issue though. Neither raw transforms nor label transforms must inherit from kornia augmentations. The only requirement for them is that they are callable and take a single argument (raw data /  labels).

**Comment by FynnBe** [comment]
Ah, I see. So how do you combine transformations? I thought this should be done with the `KorniaAugmentationPipeline` , but that failed due to it acting on multiple tensors rather than an indiviudal one. So if I want to mix/concatenate transforms/augmentation I should just make a callable transform class and put everything in there, that I need?

**Comment by constantinpape** [comment]
> Ah, I see. So how do you combine transformations?

Chaining callables is so trivial that I haven't implemented anything special for this and either wrote a small local function or used functools.reduce.
But we could add something like `Chain` or `Concatenate` to the generic transformations.

> I thought this should be done with the `KorniaAugmentationPipeline`

That's only necessary for kornia augmenations, because it does several fancy things to manage the random state of them etc.

**Comment by FynnBe** [comment]
I think I just tried too hard to use what's already there^^
It would also be easy to write a wrapper around a simple transform to make it kornia pipeline compatible... 
We can leave this for future PRs

**Review by constantinpape — APPROVED** [review]
